gpkit/nomials/nomial_math.py

```diff
@@ -693,12 +693,20 @@ def as_posyslt1(self):
 
     def _fill_default_x0(self, x0, varkeys):
         """For all keys in varkeys, updates x0 with default values and
-        substitutions. Returns x0."""
+        substitutions. Returns x0.
+
+        Order of precedence for x0 default:
+            - substitution value
+            - x0 value
+            - sp_init value
+            - 1.0
+        """
         if x0 is None:
-            x0 = {vk: vk.descr["sp_init"] for vk in varkeys
-                  if "sp_init" in vk.descr}
```

Comment:
Note that `sp_init` was not getting used for partial subs - I think this was probably a bug